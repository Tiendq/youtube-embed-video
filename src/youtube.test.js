import React from "react";
import renderer from "react-test-renderer";
import YoutubeEmbedVideo from "./youtube";

test("Default settings, need only video ID", () => {
  const component = renderer.create(
    <YoutubeEmbedVideo videoId="RnDC9MXSqCY" />
  );

  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test("Enhanced privacy mode and autoplay", () => {
  const component = renderer.create(
    <YoutubeEmbedVideo videoId="RnDC9MXSqCY" enhancedPrivacy={true} autoplay={true} />
  );

  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test("Largest size without suggestions and media controls", () => {
  const component = renderer.create(
    <YoutubeEmbedVideo width={1280} height={720} videoId="RnDC9MXSqCY" suggestions={false} controls={false} />
  );

  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
