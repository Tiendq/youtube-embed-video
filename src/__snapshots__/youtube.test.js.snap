exports[`test Default settings, need only video ID 1`] = `
<iframe
  allowFullScreen={true}
  frameBorder="0"
  height={315}
  src="https://www.youtube.com/embed/RnDC9MXSqCY?autoplay=0"
  width={560} />
`;

exports[`test Enhanced privacy mode and autoplay 1`] = `
<iframe
  allowFullScreen={true}
  frameBorder="0"
  height={315}
  src="https://www.youtube-nocookie.com/embed/RnDC9MXSqCY?autoplay=1"
  width={560} />
`;

exports[`test Largest size without suggestions and media controls 1`] = `
<iframe
  allowFullScreen={true}
  frameBorder="0"
  height={720}
  src="https://www.youtube.com/embed/RnDC9MXSqCY?autoplay=0&rel=0&controls=0"
  width={1280} />
`;
